src/main/java/org/jboss/staxmapper/Namespace.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * JBoss, Home of Professional Open Source.
+ * Copyright 2023, Red Hat, Inc., and individual contributors
+ * as indicated by the @author tags. See the copyright.txt file in the
+ * distribution for a full listing of individual contributors.
+ *
+ * This is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as
+ * published by the Free Software Foundation; either version 2.1 of
+ * the License, or (at your option) any later version.
+ *
+ * This software is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this software; if not, write to the Free
+ * Software Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA
+ * 02110-1301 USA, or see the FSF site: http://www.fsf.org.
+ */
+
+package org.jboss.staxmapper;
+
+/**
+ * Defines a versioned XML namespace.
+ * @author Paul Ferraro
+ */
+public interface Namespace<N extends Namespace<N>> {
```

Comment:
This would signify an object which is-a namespace, okay so far, I guess this is an object representation of the information contained in the namespace URI?

Reply:
The purpose is to encapsulate the info comprising the namespace URI, and provide a means to compare namespaces for use by an XMLElementReader.
The general expectation is that user would be implement this interface via an enum, which would represent the set of namespaces supported by a given reader.